### ISSUE TITLE
Other datasets cannot gradient descent

### ISSUE THREAD

**Issue by FengheTan9** [JCruan519/EGE-UNet#2]
Hello, we use [the BUSI dataset](https://www.kaggle.com/aryashah2k/breast-ultrasound-images-dataset) similar to single-object segmentation where the gradient cannot descent.

**Comment by JCruan519** [comment]
Hello, the main focus of our model is to design an extremely small model for skin lesion segmentation, as indicated by the title of the paper. Therefore, we did not train or test the model on other datasets. Furthermore, in future work, we may improve and expand the modules proposed in EGE-UNet to adapt them to other broader medical image datasets.

**Comment by zhaoyueyi** [comment]
Hi, I also found this issue while using my own dataset, and I found all my conv-layer's weight became NaN after one loss backward.
Then `with torch.autograd.detect_anomaly()` said groupConvBackward output NaN.
After some useless debugging from my guesses, I found the dataset `isic*` used in the paper is kinda special. All its samples include positive area, which means there are always two classes in each mask.
So I tentatively deleted all full-positive & full-negative samples of my dataset, then it works!
But it's truly an issue that we will meet full-positive or full-negative sample in practical usage, I don't know if it can solve this situaltion.
Thanks :)

**Comment by zhaoyueyi** [comment]
I filtered my dataset again after, and I found that conclusion above is not quite correct.
It's dirty data leading to this issue. My dataset include some edge clips from each complete samples on a black canvas with angles, and these clips are full-zero matrix.
I took back to my dataset the rest of those samples I deleted before, after deleting full-zero input samples, and it also works.
